satellytes/satellytes.com
# feat: add structured data for job postings

## What is included?
This PR adds [Structured Data](https://developers.google.com/search/docs/advanced/structured-data/job-posting#technical-guidelines) to our Job Postings on the Career Detail Pages, so that they can be found via Google Jobs.




## Which Data is inserted?

<details>
<summary>Inserted Data as Json</summary>

```json
{
  "@context":"https://schema.org/", 
  "@type":"JobPosting",
  "title":"Staff Software Engineer",
  "datePosted":"2022-01-11T12:28:42.000Z",
  "description":"description with html",
  "identifier":
  {
      "@type":"PropertyValue",
      "name":"Satellytes Digital Consulting GmbH",
      "value":"338504-41230"
  },
  "hiringOrganization":
  {
      "@type":"Organization",
      "name":"Satellytes Digital Consulting GmbH",
      "sameAs":"http://www.satellytes.com",
      "logo":"https://we-are-hiring.cdn.personio.de/logos/41230/social/6ca8f29950517ac39fcae0d6545a1fe6.png"
  },
  "jobLocation":
  {
      "@type":"Place",
      "address":
       {
            "@type":"PostalAddress",
            "streetAddress":"Sendlinger Straße 52",
            "addressLocality":"München",
            "addressRegion":"Bayern",
            "postalCode":"80331",
            "addressCountry":"DE"
       }
  },
  "jobLocationType":"TELECOMMUTE",
  "employmentType":"FULL_TIME",
  "directApply":true
}
```

[All possible data tags](https://developers.google.com/search/docs/advanced/structured-data/job-posting#structured-data-type-definitions)

</details>





#### Google Rich Search Test
Currently the [Google Rich Search Engine Test Results](https://search.google.com/test/rich-results/result/r%2Fjobs?id=N_W_P2SsDvHNp0kyaILqQQ) are missing `validThrough` and `baseSalary`.

But since they are both optional and we dont have their values, I suggest to leave them out for now.



#### Schema.org Validator
In Addition, the schma.org structured data [validator](https://validator.schema.org/#url=https%3A%2F%2Fsatellytescommain21751-feataddstructureddate.gtsb.io%2Fcareer%2Fstaff-software-engineer%2F) is fine with the provided data






## How to Test
Since the Gatsby Previews dont get crawled by Google it is not possible to check if the structured Data works as expected.
This will be possible once we deploy this on satellytes.com and triggered a new crawl for satellytes.com.
